src/components/todoItem/todoItem.tsx

```diff
@@ -0,0 +1,256 @@
+import classNames from 'classnames';
+import { useRef, useEffect } from 'react';
+import { useTodos } from '../../context/todoProvider';
+import { Todo } from '../../types/Todo';
+import { ErrorType } from '../../types/Error';
+import { toggleStatus } from '../../api/todos';
+
+type Props = {
+  task: Todo;
+  handleDeleteClick: (id: number) => void;
+};
+
+export const TodoItem = ({ task, handleDeleteClick }: Props) => {
+  const {
+    deletingTask, setError, todos, setTodos,
+    togglingId, setTogglingId, isEdited, setIsEdited,
+    isAddingTask, setIsAddingTask,
+  } = useTodos();
+
+  const inputEditRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (isEdited && inputEditRef.current) {
+      inputEditRef.current.focus();
+    }
+  }, [isEdited]);
+
+  const toggleTodo = (id: number) => {
+    const todo = todos.find(el => el.id === id);
+    const currentTogglingId = [...togglingId, id];
+
+    setTogglingId(currentTogglingId);
+
+    toggleStatus(id, { completed: !todo?.completed })
+      .then((data) => {
+        const updatedTodos = todos.map(el => (el.id === id ? data : el));
+
+        setTodos(updatedTodos);
+      })
+      .catch(() => setError(ErrorType.update))
+      .finally(() => setTogglingId([]));
+  };
+
+  const handleTitleChange = (id: number) => () => {
+    setIsEdited(id);
+  };
+
+  const handleSaveEdited = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setIsAddingTask(true);
+
+    toggleStatus(isEdited!, { title: inputEditRef.current?.value })
+      .then(data => {
+        const copy = [...todos];
+        const index = todos.findIndex(el => el.id === isEdited);
+
+        copy[index] = data;
+
+        setTodos(copy);
+      })
+      .catch(() => setError(ErrorType.update))
+      .finally(() => {
+        setIsAddingTask(false);
+        setIsEdited(null);
+      });
```

Comment:
I think I'd move this logic to some method (e.g. updateTodo) in the context provider since it's repeated later onBlur, however works like that as well